### ISSUE TITLE
Setting minY for Line Chart Causes Rendering outside of Container

### ISSUE THREAD

**Issue by herrchmafi** [imaNNeoFighT/fl_chart#870]
** Don't make a duplicate issue.
You can search in issues to make sure there isn't any already opened issue with your concern.

**Describe the bug**
Structurally, I have a `LineChart` within a `Container` that's used to constrain the chart's dimensions. When supplying a `minY` to `LineChartData`, the chart display extends outside the bounds of the `Container`. Normally, I would expect the chart to scale to the size of the container in all cases.

**To Reproduce**

Code snippet:
<img width="471" alt="Screen Shot 2022-01-06 at 10 19 36 AM" src="https://user-images.githubusercontent.com/10555443/148432023-71cc80e0-39f7-470c-a1ee-cbaaa15dad3d.png">

The y values supplied range between 35,000 to 65,000

**Screenshots**

Without specifying a `minY`:
<img width="331" alt="Screen Shot 2022-01-06 at 10 19 59 AM" src="https://user-images.githubusercontent.com/10555443/148432333-ccef8f98-2b93-46ab-8a57-af33c2d124a2.png">

Specifying a `minY`:
<img width="330" alt="Screen Shot 2022-01-06 at 10 18 29 AM" src="https://user-images.githubusercontent.com/10555443/148432360-a2ca098e-49cb-48de-b3c7-719aad86eea2.png">

**Versions**
 - which version of the Flutter are you using?
2.5.3
 - which version of the FlChart are you using?
0.40.6

**Comment by imaNNeoFighT** [comment]
Can you explain more about your expectation?
Are you expecting to cut the line in this scenario?

**Comment by herrchmafi** [comment]
Right! So, the graph has a specified area via the container that should be above the buttons. My expectation would be that the drawn chart doesn't go outside of the bounds and instead scales accordingly.

**Comment by imaNNeoFighT** [comment]
We can't scale down the chart.
For example you have these points: [0, -2], [1, -3]. And you set `minY = 0`, `maxY = 2`. 
How can we scale the chart? You need to pass a `minY` that works with your data.

BTW I think we need to cut the chart by default.

**Comment by Harishwarrior** [comment]
Facing the same problem.

**Comment by perttikasanenhitseed** [comment]
I faced the same problem. It would be good to be able to cut the lines outside the min and max.

**Comment by chitgoks** [comment]
The problem with this is that if the maxY or minY is the same as the dot value, the dot rendering gets cut off in half. the only way to fix this is to manually add values to min max x,y so the dots can be fully seen.
